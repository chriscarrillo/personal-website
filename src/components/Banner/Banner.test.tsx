import {describe, expect, it} from '@jest/globals'
import {render} from '@testing-library/react'
import React from 'react'
import renderer from 'react-test-renderer'
import {Banner} from '.'

describe('<Banner />', () => {
  it('should render as expected', () => {
    const tree = renderer.create(<Banner />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('should show image', async () => {
    const {findByAltText, findByTestId} = render(<Banner />)

    await expect(findByTestId('avatar')).toBeTruthy()
    await expect(findByAltText('Avatar')).toBeTruthy()
  })
})
