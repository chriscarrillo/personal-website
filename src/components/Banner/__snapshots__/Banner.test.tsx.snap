// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Banner /> should render as expected 1`] = `
<div
  name="home"
>
  <div
    className="
        banner
        bg-dark
        w-100
        d-flex
        flex-column
        justify-content-center
        align-items-center
        text-center
        text-light"
  >
    <img
      alt="Avatar"
      className="pt-3"
      data-testid="avatar"
      src="avatar.svg"
    />
    <h1
      className="pt-2 pb-1 mb-0 family-semi text-uppercase"
    >
      Chris Carrillo
    </h1>
    <h2
      className="h6 pb-3"
    >
      Software Engineer - Coffee Lover - Enneagram Enthusiast
    </h2>
  </div>
</div>
`;
